app/javascript/App/bundles/common/components/CurrencySelector.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import { Select, createStyles, withStyles } from '@material-ui/core'
+
+interface Props {
+  initialCurrency?: string
+  isManagedState?: boolean
+  onChange?: (selected: string) => void
+  value?: string
+  classes: any
+}
+
+interface State {
+  selectedCurrency: string
+}
+
+const styles = (theme) =>
+  createStyles({
+    root: {
+      color: '#fff',
+    },
+    icon: {
+      color: '#fff',
+    },
+  })
+
+class CurrencySelector extends React.Component<Props, State> {
+  constructor(props) {
+    super(props)
+
+    const managedCurrency = props.isManagedState
+      ? props.value
+      : props.initialCurrency
+    const selectedCurrency = managedCurrency || CURRENCIES[0]
+    this.state = {
+      selectedCurrency,
+    }
+  }
+
+  public handleChange = (e) => {
+    e.preventDefault()
+    e.stopPropagation()
+    const { onChange, isManagedState } = this.props
+    const value = e.target.value
+
+    if (onChange) {
+      onChange(value)
+    }
+    if (!isManagedState) {
+      this.setState({
+        selectedCurrency: value,
+      })
+    }
+  }
+
+  public componentDidUpdate(prevProps, prevState) {
+    const { isManagedState, value } = this.props
+    if (
+      isManagedState &&
+      _.isString(value) &&
```

Comment:
When it's undefined. I was thinking it would be better to use positive type checks rather than !_.isUndefined() especially since typescript it static type checking.

Reply:
I guess if something weird happens runtime and it does come in as something other than `string`, would you prefer it to fail horribly or ignore the value?